File: qt/gen_qbindingstorage.go
package qt

import (
	"unsafe"
)

type QBindingStatus struct {
	h          uintptr
	isSubclass bool
}

type QBindingStorage struct {
	h          uintptr
	isSubclass bool
}

// NewQBindingStorage constructs a new QBindingStorage object.
func NewQBindingStorage() *QBindingStorage {

	ret := newQBindingStorage(QBindingStorage_new())
	ret.isSubclass = true
	return ret
}

func (this *QBindingStorage) IsEmpty() bool {
	return (bool)(QBindingStorage_IsEmpty(this.h))
}

func (this *QBindingStorage) IsValid() bool {
	return (bool)(QBindingStorage_IsValid(this.h))
}

func (this *QBindingStorage) RegisterDependency(data *QUntypedPropertyData) {
	QBindingStorage_RegisterDependency(this.h, data.cPointer())
}

File: qt/gen_qbindingstorage.cpp
// +build ignore

#include <QBindingStatus>
#include <QBindingStorage>
#include <QUntypedPropertyData>
#include <qbindingstorage.h>
#include "gen_qbindingstorage.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<_GUID*>( self );
	} else {
		delete self;
	}
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}

void QBindingStatus_Delete(QBindingStatus* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QBindingStatus*>( self );
	} else {
		delete self;
	}
}

QBindingStorage* QBindingStorage_new() {
	return new QBindingStorage();
}

bool QBindingStorage_IsEmpty(QBindingStorage* self) {
	return self->isEmpty();
}

bool QBindingStorage_IsValid(const QBindingStorage* self) {
	return self->isValid();
}

void QBindingStorage_RegisterDependency(const QBindingStorage* self, QUntypedPropertyData* data) {
	self->registerDependency(data);
}

void QBindingStorage_Delete(QBindingStorage* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QBindingStorage*>( self );
	} else {
		delete self;
	}
}

